### PR TITLE
feat: expose cleanup_metadata in Python

### LINKED ISSUE #1768: Expose cleanup_metadata or create_checkpoint_from_table_uri_and_cleanup to the Python API
# Description
When utilizing the `table.create_checkpoint()` method in Python, direct access to the `cleanup_metadata()` function is currently unavailable. This limitation results in an increased number of log entries. In Spark, this feature is integrated as a step at the end of the method (as evident in the [Spark implementation](https://github.com/delta-io/delta/blob/8639c411890a5c77386f04e2282fcf4caa401eff/spark/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala#L303)). However, it's conspicuously absent from the Python API.

In the Rust implementation (as seen in the [Rust implementation](https://github.com/delta-io/delta-rs/blob/a67140c6b13a6f820f4879f31eb927d140327746/rust/src/protocol/checkpoints.rs#L92C14-L92C58)), there's a `cleanup_metadata` and also `create_checkpoint_from_table_uri_and_cleanup` method, but it's not called at the end of the `create_checkpoint` process.

There is another [Pull Request (PR)](https://github.com/delta-io/delta-rs/pull/1728) that addresses the creation of a checkpoint both before and after the vacuum operation. However, the absence of the `cleanup_metadata` method during the checkpoint creation process results in the metadata directory remaining uncleared.

**Use Case**
Cleaning up the Metadata Directory After Checkpoint Creation or Optimization

**Related Issue(s)**
#1728

### PR DESCRIPTION
**Pull request by r3stl355**
# Description
- exposed existing `cleanup_metadata` from Rust in Python bindings
- added a Python unit test

# Related Issue(s)
- closes #1768

### CODE REVIEW

**Review by wjones127 — APPROVED** [review]
One minor ask, otherwise looks good!

File: python/deltalake/table.py [hunk 1]
```diff
@@ -799,6 +799,13 @@ def update_incremental(self) -> None:
     def create_checkpoint(self) -> None:
         self._table.create_checkpoint()
 
+    def cleanup_metadata(self) -> None:
+        """
+        Delete expired log files before current version from table. The table log retention is based on
+        the `configuration.logRetentionDuration` value.
```
[the remark this answers is not in the document]
Comment (r3stl355) [reply python/deltalake/table.py]: Done. I did not include that initially as I am not a huge fan of including info like default values defined elsewhere in comment as those values may change independently, but at the same time I recognise that not providing that info leave a bit of a gap so no solution is perfect.

**Review by wjones127 — APPROVED** [review]

**Comment by r3stl355** [comment]
Hey @wjones127, the benchmark build is failing but I can't see how that can be related to my PR, any thoughts?

**Comment by r3stl355** [comment]
Thank you for merging this!!

**Pull request merged**